crates/bevy_app/src/event.rs

```diff
@@ -56,11 +56,23 @@ enum State {
 }
 
 /// An event collection that represents the events that occurred within the last two
-/// [Events::update] calls. Events can be cheaply read using an [EventReader]. This collection is
-/// meant to be paired with a system that calls [Events::update] exactly once per update/frame.
-/// [Events::update_system] is a system that does this. [EventReader]s are expected to read events
-/// from this collection at least once per update/frame. If events are not handled within one
-/// frame/update, they will be dropped.
+/// [`Events::update`] calls.
+/// Events can be written to in parallel using an [`EventWriter`]
+/// and are typically cheaply read using an [`EventReader`].
+///
+/// Each event can be consumed by multiple systems, in parallel,
+/// with consumption tracked by the [`EventReader`] on a per-system basis.
+///
+/// This collection is meant to be paired with a system that calls
+/// [`Events::update`] exactly once per update/frame.
+///
+/// [`Events::update_system`] is a system that does this, typically intialized automatically using
+/// [`AppBuilder::add_event`]. [EventReader]s are expected to read events from this collection at
+/// least once per loop/frame.  
+/// Events will persist across a single frame boundary and so ordering of event producers and
+/// consumers is not critical (although poorly-planned ordering may cause accumulating lag).
+/// If events are not handled by the end of the frame after they are updated, they will be
```

Comment:
Nit: "by the end of the frame after they are created"

Reply:
Eh actually thats _only_ true when updates happen every frame. I guess your phrasing is more correct in the general case.